channeldb/codec.go

```diff
@@ -178,7 +179,17 @@ func WriteElement(w io.Writer, element interface{}) error {
 		}
 
 	case lnwire.Message:
-		if _, err := lnwire.WriteMessage(w, e, 0); err != nil {
+		var msgBuf bytes.Buffer
+		if _, err := lnwire.WriteMessage(&msgBuf, e, 0); err != nil {
+			return err
+		}
+
+		msgLen := uint16(len(msgBuf.Bytes()))
+		if err := WriteElements(w, msgLen); err != nil {
+			return err
+		}
+
+		if _, err := w.Write(msgBuf.Bytes()); err != nil {
```

Comment:
By wire I mean as it's written to our "socket". That "socket" then wraps the wire message in an encrypted transport. The wire message itself doesn't include the length prefix, the transport actually does. 

The two options for length encoding are either varint, or just a `uint16`. I've gone with the `uint16` here.

Reply:
Ok, so it happens to be the same, but we could just as well have used any of the other ways to encode the length, like the length encoding in our tlv format? Or even encode the message in single record tlv format, so that we can add more fields in the future without relying on EOF?

Reply:
On the wire level, we still need to rely on the EOF, or do incremental byte accounting to make sure we don't read more than we need to. Here in the database, we can use any of those formats. Doing this migration now means we can actually use TLV within the DB for any extensions in the future.

Reply:
Suppose you want to add a tlv block at the end (after the message that is migrated in this pr) in a future pr, then you still need either a migration or an EOF check to see if the tlv block is there or not?

What I suggested is to introduce the tlv block already in this PR. Not converting all the existing fields, but just replace the message by a single-record tlv block.

Existing format: "MESSAGE"
This pr: 7 | "MESSAGE" (prepend length)
Suggestion: 0 | 7 | "MESSAGE" (make it tlv already now with the message being type 0, so that more data can be added later)

This then does need to be moved to the very end of the bbolt value, otherwise another length field is needed.

But yes, it is just a suggestion. Can also see that we don't want to extend it like that now.